### DISCUSSION TITLE
KPN IPTV with VDSL

### DISCUSSION THREAD

**Discussion by coldburn89** [fabianishere/udm-iptv#158] [General]
Hi Fabian,

Maybe this question is out of your reach, but since your script works for Fiber, maybe you also have info on VDSL.

I used to get IPTV working through a VLAN-only IPTV on my 2 KPN set-top boxes, see attached image. Since the last update of the UDM, from 1.11.4 to 1.12.xx it no longer works. KPN's well-known 400 error code is playing up and this seems to be an IGMP issue. For now I did a rollback and re-flashed the 1.11.4 and the problem is solved. Weird right?

Do you have a solution for the 1.12.xx version to get KPN IPTV working? Is this a bug in the firmware or is the below not a good configuration? In principle, the following (with VLAN Only) should be a good solution for looping through iPTV to the set-top boxes.

I uploaded the config i am currently using. I also found your script but when i execute the first steps and configure PPOEE, i shut myself out of the system. For some reason i can't get it configured. The thing is, will i create issues when i install your udm-iptv package?

Weird thing is -> in 1.11.4 IPTV works like a charm and in 1.12.xx its broken.

Thanks in advance!

Regards Aaron

![image](https://user-images.githubusercontent.com/103142365/183383259-a20a5137-bcf4-46ad-abb8-0a92c0e034c3.png)
![image](https://user-images.githubusercontent.com/103142365/183383290-3ae2e089-9a2a-4262-8216-410f2c397b23.png)
![image](https://user-images.githubusercontent.com/103142365/183383302-f93265c5-1bca-4192-aeeb-365db1da3e77.png)

**Comment by fabianishere** [comment]
This could be related to IGMP snooping, which has been added to the UDM (Pro) switch since 1.12. Could you try disabling it?
```bash
swconfig dev switch0 set igmp_snooping 0
```

Alternatively, it might be easier just to setup routed IPTV using this project. Perhaps it is unclear in the guide, but a fiber connection is not strictly necessary for this project to work. However, since you are using the Experiabox in front of the UDM, you need to set `IPTV_WAN_VLAN` to zero (that is, IPTV traffic is carried over the WAN port without VLAN).

**Reply by coldburn89** [reply]
Indeed, i think there is the confusion for alot of people that don't have fiber. But it makes no sense, because i run a vlan-only network for IPTV. 

> Alternatively, it might be easier just to setup routed IPTV using this project. Perhaps it is unclear in the guide, but a fiber connection is not strictly necessary for this project to work. However, since you are using the Experiabox in front of the UDM, you need to set IPTV_WAN_VLAN to zero (that is, IPTV traffic is carried over the WAN port without VLAN).

Does this affect the coding of your project? Like you said, change vlan to 0. But are there more changes i should make?

Regards Aaron

**Reply by fabianishere** [reply]
Does disabling IGMP Snooping (with the command I gave) work for your setup?  

If you want to use this project, just have one connection from the Experiabox to the WAN port of the UDM and configure the script to use VLAN 0.

You can then have the STB on your local LAN or a separate VLAN if you prefer.

**Reply by coldburn89** [reply]
Hi,

Didn't tried it yet, as it's a big hassle to fix it. First, i need to update to the latest firmware. When it doesn't work I need to recover everything. I will try this next weekend :)

**Reply by coldburn89** [reply]
Hi Fabian, i tried several things after upgrading the UDM to FW 1.12.22:

1. Removed 2nd UTP cable for IPTV (from EB to UDM) and disabled IGMP from IPTV vlan = No succes. Also when i performed "swconfig dev switch0 set igmp_snooping 0" it gave an error that the command was not correct.
2. Switched from Vlan-only profile to VLAN with DHCP = No succes (same with IGMP enabled)
3. Runned your script, selected "custom" and configured vlan0 @wan and selected the LAN. Also i configured the STB on local LAN = no success.
4. When reverting to Vlan-only, it works for 5 min and then it stops. After that i need to reboot the STB and then it works again for 5 mins.
5. So after all, IPTV is still broken after the 1.12.22 update of the UDM and running your script. 

With all this crap from KPN i am considering another provider, as this is just bullshit.

**Reply by fabianishere** [reply]
What error did you get when running the command I proposed? If you use that approach, you should not make any changes to your network, other than upgrading to 1.12.

If you choose the other approach where you connect the UDM to the EB with only a single cable, please share the output of the diagnostic tool. Otherwise I cannot be of help to you.

**Reply by coldburn89** [reply]
Thanks for the reply! Unfortunately I ended up with an UTP cable from IPTV to EB. This way i will never have any issues when upgrading FW.
